src/llfs/no_outgoing_refs_cache.cpp

```diff
@@ -0,0 +1,75 @@
+//#=##=##=#==#=#==#===#+==#+==========+==+=+=+=+=+=++=+++=+++++=-++++=-+++++++++++
+//
+// Part of the LLFS Project, under Apache License v2.0.
+// See https://www.apache.org/licenses/LICENSE-2.0 for license information.
+// SPDX short identifier: Apache-2.0
+//
+//+++++++++++-+-+--+----- --- -- -  -  -   -
+#include <llfs/no_outgoing_refs_cache.hpp>
+
+namespace llfs {
+
+//==#==========+==+=+=++=+++++++++++-+-+--+----- --- -- -  -  -   -
+//
+NoOutgoingRefsCache::NoOutgoingRefsCache(const PageIdFactory& page_ids) noexcept
+    : page_ids_{page_ids}
+    , cache_(this->page_ids_.get_physical_page_count().value())
+{
+}
+
+//==#==========+==+=+=++=+++++++++++-+-+--+----- --- -- -  -  -   -
+//
+void NoOutgoingRefsCache::set_page_state(PageId page_id,
+                                         batt::BoolStatus new_has_outgoing_refs_state) noexcept
```

Comment:
As discussed offline, this param should change to bool (strong typedef), because kUnknown is invalid here.

Reply:
Fixed